packages/squiggle-lang/src/ast/peggyParser.peggy

```diff
@@ -58,13 +58,38 @@ decorator
       return h.nodeDecorator(name, args ?? [], location());
     }
 
+typeSignature
+  = _ ":" _ body:infixType
+    {
+      return h.nodeTypeSignature(false, body, location());
+    }
+  / body:implicitType
+    {
+      return h.nodeTypeSignature(true, body, location());
+    }
+
+infixType
+  = head:identifier tail:(@typeMultiplicativeOp @identifier)*
+    { return h.makeInfixTypeChain(head, tail, location()); }
+  typeMultiplicativeOp "operator" = '*' / '/'
+
+implicitType
+  = ""
+    { return h.nodeImplicitType(location()); }
```

Comment:
Making `typeSignature` optional is non-trivial because an empty type has actual semantics and it's useful to have an `ImplicitType` node to handle this. You could also handle it by checking `NodeLetStatement.typeSignature === undefined` but (1) this would require a special case and (2) my preference is to be explicit about when types are implicit.

Reply:
> You could also handle it by checking `NodeLetStatement.typeSignature === undefined` but (1) this would require a special case and (2) my preference is to be explicit about when types are implicit.

I don't understand why undefined check is a problem (also you can use `!NodeLetStatement.typeSignature`, since `typeSignature` is an object and so it's truthy); you either have to do that or check `typeSignature.isImplicit` everywhere, and the latter is just more verbose and less memory-efficient.

Reply:
I will change this to work like you suggested, but since it's just a cosmetic change, I'll do it after I get all the functionality working.

My latest commit introduced a similar pattern [here](https://github.com/quantified-uncertainty/squiggle/pull/3323/files#diff-c551d40e63d35b3be13c121195361b9e75f4b55190ffc1db82700fb5f8320f6eR61-R65) where I have an explicitly-marked object that could just be `undefined`, I can change that too if you want.